src/components/Footer.tsx

```diff
@@ -0,0 +1,68 @@
+import cn from 'classnames';
+import { CompletedFilter } from '../types/CompletedFilter';
+
+type Props = {
+  uncompletedTodosAmount: number;
+  hasCompletedTodos: boolean;
+  completedFilter: CompletedFilter;
+  removeCompletedTodos: () => void;
+  setCompletedFilter: (filter: CompletedFilter) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  uncompletedTodosAmount,
+  hasCompletedTodos,
+  completedFilter,
+  removeCompletedTodos,
+  setCompletedFilter,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${uncompletedTodosAmount} item${uncompletedTodosAmount > 1 ? '\'s' : ''} left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={cn('filter__link',
+            { selected: completedFilter === CompletedFilter.All })}
+          onClick={() => setCompletedFilter(CompletedFilter.All)}
+        >
+          All
+        </a>
+
+        <a
+          data-cy="FilterLinkActive"
+          href="#/active"
+          className={cn('filter__link',
+            { selected: completedFilter === CompletedFilter.Active })}
+          onClick={() => setCompletedFilter(CompletedFilter.Active)}
+        >
+          Active
+        </a>
+        <a
+          data-cy="FilterLinkCompleted"
+          href="#/completed"
+          className={cn('filter__link',
+            { selected: completedFilter === CompletedFilter.Completed })}
+          onClick={() => setCompletedFilter(CompletedFilter.Completed)}
+        >
+          Completed
+        </a>
+      </nav>
+
+      {hasCompletedTodos && (
+        <button
+          data-cy="ClearCompletedButton"
+          type="button"
+          className="todoapp__clear-completed"
+          onClick={() => removeCompletedTodos()}
+        >
+          Clear completed
+        </button>
+      )}
```

Comment:
This condition brock UI, do not remove btn, u could make it disabled

Reply:
don't understand what you want.
in our task - I should do visible/invisible button, not disabled

or are you talking about styles and I should do something in CSS?
help me, please.

Reply:
I think I understand
I changed the style in the third part